src/shell-window-tracker.c

```diff
@@ -510,10 +510,16 @@ get_app_for_window (ShellWindowTracker    *tracker,
   {
     if (tracker->coding_app) {
       g_hash_table_insert (tracker->builder_to_app, window, tracker->coding_app);
```

Comment:
To me it feels like this function should be indepotent (`get_app_for_window`) and is callable from JS, so calling `g_hash_table_insert` here is confusing. More confusing is that the watchdog id is removed at this point (assumes that `tracker->watchdog_id > 0`?)